doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md

```diff
@@ -0,0 +1,219 @@
+
+```shell
+$ kubectl ray log ray-cluster-kuberay
+
+No output directory specified, creating dir under current directory using cluster name.
+Downloading file ./ for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./old/ for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./gcs_server.out for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./gcs_server.err for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./events/ for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./events/event_GCS.log for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./events/event_AUTOSCALER.log for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./events/event_RAYLET.log for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./debug_state_gcs.txt for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./monitor.out for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./monitor.err for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./ray_client_server.out for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./ray_client_server.err for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./dashboard.err for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./dashboard.log for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./monitor.log for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./raylet.out for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./raylet.err for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./log_monitor.err for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./debug_state.txt for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./log_monitor.log for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./dashboard_agent.log for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./agent-424238335.out for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./agent-424238335.err for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./runtime_env_agent.out for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./runtime_env_agent.err for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./runtime_env_agent.log for Ray Head ray-cluster-kuberay-head-qkrv8
+Downloading file ./nsight/ for Ray Head ray-cluster-kuberay-head-qkrv8
+```
+
+## Submit a Ray job
+
+This is a wrapper around the `ray job submit` command. The plugin will automatically forward the ports to the Ray cluster and submit the job.
+
+Assume that under the current directory, you have a file named `task.py`.
+
+```python
+import ray
+ray.init(address="auto")
+
+@ray.remote
+def f(x):
+    return x * x
+
+futures = [f.remote(i) for i in range(4)]
+print(ray.get(futures)) # [0, 1, 4, 9]
+```
+
+And a file named `ray-job.usermode.yaml`
+
+```yaml
+apiVersion: ray.io/v1
+kind: RayJob
+metadata:
+  name: rayjob-usermode
+spec:
+  submissionMode: UserMode
```

Comment:
Given we're about to change this for v1.3, should we remove any examples of `kubctl ray job submit`? I'm worried users will try with UserMode with 1.2 and then we'll break them in 1.3

Reply:
Done in https://github.com/ray-project/ray/pull/47851/commits/c87ec4abd47f7fd52c43b688556d854f3664e886. `kubectl ray job submit` example is completely removed.